### PR TITLE
Nonmovable Parameter class

### PR DESCRIPTION
**Pull request by odashi**
This PR modifies:
* Make `Parameter` class non-movable.
* Make `Parameter::load()` function non-static.
* Add `Parameter::init()` function to allow in-place initialization.

This PR does not pass CI. TODO after this PR is:
* Fix Python interfaces and tests.

### CODE REVIEW

**Comment by odashi** [comment]
@vbkaisetsu, please check out this PR into other working branch and fix Python interface aboud Parameter.

File: primitiv/parameter.h [hunk 1]
```diff
@@ -65,27 +44,57 @@ class Parameter : mixins::Noncopyable<Parameter> {
...
+   * Loads parameters from specified file.
+   * @param path File path to load parameters.
+   * @param with_stats Whether or not to load all additional statistics as well
+   *                   as parameter values if the file has them.
+   * @param device The device object to manage internal memory.
    */
-  void reset_value(const std::vector<float> &value);
+  void load(
+      const std::string &path,
+      bool with_statis = true,
```
Comment (vbkaisetsu) [inline primitiv/parameter.h]: Spelling error: `with_statis` -> `with_stats`

File: primitiv/parameter.h [hunk 2]
```diff
@@ -65,27 +44,57 @@ class Parameter : mixins::Noncopyable<Parameter> {
...
+  /**
+   * Initializes the Parameter object.
+   * @param shape The shape of the parameter. The batch size should be 1.
+   * @param init An Initializer object.
+   * @param device The device object to manage internal memory.
+   */
+  void init(
+      const Shape &shape,
+      const Initializer &initializer,
+      Device &device = Device::get_default());
```
Comment (vbkaisetsu) [inline primitiv/parameter.h]: Some functions take `shape` and `initializer` in this order in C++, whereas python-primitiv can also take an `np.ndarray` argument without a `shape` argument.
Therefore, it is very useful if the order is `initializer` -> `shape`.

Do you have a plan to change this order?

Comment (odashi) [reply primitiv/parameter.h]: It is better to move this topic to other issue.

File: primitiv/parameter.h (hunk 1 above)
Comment (odashi) [reply primitiv/parameter.h]: Fixed.

**Comment by vbkaisetsu** [comment]
I fixed Python interface. Please see #58

File: primitiv/parameter.h (hunk 2 above)
Comment (odashi) [reply primitiv/parameter.h]: See #57.

Comment (vbkaisetsu) [reply primitiv/parameter.h]: Okay. For now, I use keyword arguments with the same order.

**Pull request merged**